File: ChartLib/src/main/java/com/yxc/chartlib/listener/SimpleItemGestureListener.java
package com.yxc.chartlib.listener;

import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

import com.github.mikephil.charting.data.BarEntry;
import com.yxc.fitness.chart.entrys.RecyclerBarEntry;

/**
 * @author yxc
 * @since  2019/4/24
 */
public abstract class SimpleItemGestureListener implements OnItemGestureListener<RecyclerBarEntry> {
    @Override
    public void onItemClick(View view, int position) {

    }

    @Override
    public void onLongItemClick(View view, int position) {

    }

    @Override
    public void onItemSelected(RecyclerBarEntry barEntry, int position) {

    }


    public abstract void onItemSelected(BarEntry barEntry, int position);

    @Override
    public void onScrollStateChanged(RecyclerView recyclerView, int newState) {

    }

    @Override
    public void onScrolled(RecyclerView recyclerView, int dx, int dy) {

    }

}

File: ChartLib/src/main/java/com/yxc/chartlib/listener/SimpleItemGestureListener.kt
package com.yxc.chartlib.listener

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.yxc.fitness.chart.entrys.RecyclerBarEntry

/**
 * @author yxc
 * @since  2019/4/24
 */
open class SimpleItemGestureListener<T:RecyclerBarEntry?> : OnItemGestureListener<RecyclerBarEntry> {
    override fun onItemClick(view: View, position: Int) {}
    override fun onLongItemClick(view: View, position: Int) {}
    override fun onItemSelected(barEntry: RecyclerBarEntry?, position: Int) {}
    override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {}
    override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {}
}